src/common/structure/yaml_writer.go

```diff
@@ -63,13 +63,51 @@ func WriteYAMLFile(readFilePath string, blocks []IBlock, writeFilePath string, r
 
 	return err
 }
+func reflectValueToMap(rawMap interface{}, currResourceMap *map[string]interface{}, tagsAttributeName string) *map[string]interface{} {
+	switch rawMap.(type) {
+	case map[interface{}]interface{}:
+		rawMapCasted := rawMap.(map[interface{}]interface{})
+		for mapKey, mapValue := range rawMapCasted {
+			mapKeyString := mapKey.(string)
+			if mapKey == tagsAttributeName {
+				currResourceMap = reflectValueToMap(mapValue.(map[string]string), currResourceMap, tagsAttributeName)
+			} else {
+				switch mapValue.(type) {
+				case string:
+					(*currResourceMap)[mapKeyString] = mapValue.(string)
+					break
+				case int:
+					(*currResourceMap)[mapKeyString] = mapValue.(int)
+					break
+				case bool:
+					(*currResourceMap)[mapKeyString] = mapValue.(bool)
+					break
+				}
+			}
+		}
+		break
+	case map[string]string:
+		if _, ok := (*currResourceMap)[tagsAttributeName]; !ok {
+			(*currResourceMap)[tagsAttributeName] = make(map[string]string)
+		}
+		rawMapCasted := rawMap.(map[string]string)
+		for mapKey, mapValue := range rawMapCasted {
+			(*currResourceMap)[tagsAttributeName].(map[string]string)[mapKey] = mapValue
+		}
+		break
+	}
+	return currResourceMap
+}
 
-func getYAMLLines(rawBlock interface{}) []string {
-	yamlBytes, err := yaml.Marshal(rawBlock)
+func getYAMLLines(rawBlock map[interface{}]interface{}, tagsAttributeName string) []string {
+	var textLines []string
+	tempTagsMap := make(map[string]interface{})
+	castedRawBlock := reflectValueToMap(rawBlock, &tempTagsMap, tagsAttributeName)
```

Comment:
@tronxd Is is really necessary to call `reflectValueToMap` at all just for to call for `yaml.Marshal`?